app/controllers/bike_share_app.rb

```diff
@@ -1,3 +1,158 @@
+require_relative '../models/station'
+require_relative '../helpers/bike_share_app_helper'
+
 class BikeShareApp < Sinatra::Base
 
+  include BikeShareAppHelper
+
+  set :root, File.expand_path("..", __dir__)
+  set :method_override, true
+
+  get '/' do
+    erb :landing_page
+  end
+
+  ### Start Stations Routes ###
+  get '/station-dashboard' do
+    @stations = Station.all
+    erb :"stations/dashboard"
+  end
+
+  get '/stations' do
+    @stations = Station.all
+    erb :"stations/index"
+  end
+
+  get '/stations/new' do
+    @cities = City.all
+    erb :"stations/new"
+  end
+
+  post '/stations' do
+    params[:station][:installation_date] = format_date(params[:station][:installation_date])
+    Station.create(params[:station])
+    redirect "/stations"
+  end
+
+  get '/stations/:id' do
+    @station = Station.find(params[:id])
+    erb :"stations/show"
+  end
+
+  get '/stations/:id/edit' do
+    @station = Station.find(params[:id])
+    @cities  = City.all
+    erb :"stations/edit"
+  end
+
+  put '/stations/:id' do
+    Station.update(params[:id], params[:station])
+    redirect "/stations/#{params[:id]}"
+  end
+
+  delete '/stations/:id' do
+    Station.destroy(params[:id])
+    redirect '/stations'
+  end
+
+  ### Start Trips Routes ###
+  get '/trip-dashboard' do
+    @monthly_rides = Trip.monthly_breakdown_master
+    erb :"trips/dashboard"
+  end
+
+  get '/trips' do
+    @page  = params[:page].to_i
+    @end   = false
+    @trips = Trip.order(:start_date).offset(@page * 30).limit(30)
+    @end   = true if Trip.order(:start_date).offset((@page + 1) * 30).empty?
```

Comment:
this along with the line above is what I consider "too much logic in your controllers." we're doing a lot of calculating and specific querying here. can we put this logic in either a helper module or a method in the model?